File: benchmarks/src/python/insertion_sort.py
import time
arr = list(range(1000, 0, -1))
start = time.time() * 1000
for j in range(1, len(arr)):
    key = arr[j]
    k = j - 1
    while k >= 0 and arr[k] > key:
        arr[k + 1] = arr[k]
        k -= 1
    arr[k + 1] = key
end = time.time() * 1000
print(f"Result: {arr[0]}/{arr[-1]}")
print(f"Time: {end - start:.0f}ms")

File: benchmarks/src/lua/insertion_sort.lua
local arr = {}
for i = 1000, 1, -1 do table.insert(arr, i) end
local start = os.clock()
for j = 2, #arr do
    local key = arr[j]
    local k = j - 1
    while k >= 1 and arr[k] > key do
        arr[k + 1] = arr[k]
        k = k - 1
    end
    arr[k + 1] = key
end
local finish = os.clock()
print("Result: " .. arr[1] .. "/" .. arr[#arr])

print(string.format("Time: %.0fms", (finish - start) * 1000))

